[App extensibility] Remove no-longer-needed dependencies (@W-17053325@)

vmarta opened the pull request:
<!--- Provide a short summary of your changes in the Title field above -->

# Description

<!--- A longer summary of your changes, including: a description of the issue that you’re addressing, a list of required dependencies (if applicable), and any other relevant context. -->

# Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] **Bug fix** (non-breaking change that fixes an issue)
- [ ] **New feature** (non-breaking change that adds functionality)
- [ ] **Documentation update**
- [ ] **Breaking change** (could cause existing functionality to not work as expected)
- [ ] **Other changes** (non-breaking changes that does not fit any of the above)

> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- (step1)

# Checklists

<!--- Enter an `x` in all the boxes that apply. -->
<!--- If you’re unsure about any of these, don’t hesitate to ask. We’re here to help! -->

## General

- [ ] Changes are covered by test cases
- [ ] CHANGELOG.md updated with a short description of changes (_not_ required for documentation updates)

## Accessibility Compliance

You must check off all items in **one** of the follow two lists:

- [ ] There are no changes to UI

_or..._

- [ ] Changes were tested with a Screen Reader (iOS VoiceOver or Android Talkback) and had no issues
- [ ] Changes comply with [WCAG 2.0 guidelines levels A and AA](https://www.wuhcag.com/wcag-checklist/)
- [ ] Changes to common UI patterns and interactions comply with [WAI-ARIA best practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## Localization

- [ ] Changes include a UI text update in the Retail React App (which requires translation)

vmarta committed:
Remove no longer needed dependencies

vmarta commented on packages/pwa-kit-dev/package.json:
```diff
@@ -73,7 +73,6 @@
     "compression": "1.7.4",
     "copy-webpack-plugin": "^9.1.0",
     "cross-env": "^5.2.1",
-    "dedent": "^1.5.3",
```
This has moved to be a dependency of the other package, pwa-kit-extension-sdk.

vmarta commented on packages/pwa-kit-dev/package.json:
```diff
@@ -127,7 +124,6 @@
     "@types/node-fetch": "~2.6.3",
     "@types/validator": "~13.7.14",
     "internal-lib-build": "4.0.0-dev",
-    "memfs": "^4.11.1",
```
This has moved to be a dependency of the other package, pwa-kit-extension-sdk.

vmarta commented on packages/pwa-kit-dev/package.json:
```diff
@@ -106,8 +105,6 @@
     "source-map-loader": "^4.0.1",
     "speed-measure-webpack-plugin": "^1.5.0",
     "svg-sprite-loader": "^6.0.11",
-    "ts-loader": "^9.5.1",
-    "tsx": "^4.16.2",
```
We no longer used these, is this correct?

bendvc replied:
Correct. I think these were from the original server side extensibility implementation

vmarta commented on packages/template-retail-react-app/package.json:
```diff
@@ -41,7 +41,6 @@
     "@loadable/component": "^5.15.3",
     "@peculiar/webcrypto": "^1.4.2",
     "@salesforce/commerce-sdk-react": "3.1.0-dev",
-    "@salesforce/pwa-kit-extension-sdk": "4.0.0-dev",
```
We don't need the extension-sdk for retail-react-app, is that right?

bendvc replied:
no.. that was probably added by mistake

vmarta replied:
Hmm, CI is failing because of it. So it seems like it's really needed somehow 🤔

vmarta committed:
Bring back the dependency to pass CI test

Webpack builds require the extension-sdk, so let's include it as dev dependency.

Related initial commit: https://github.com/SalesforceCommerceCloud/pwa-kit/commit/5b29602ffd2628e7ddabe639dabb4cd35bfa2f76

vmarta committed:
Skip these failing tests for now

We plan to revisit them in another ticket.

vmarta committed:
Revert "Skip these failing tests for now"

This reverts commit 95930bffca7d085a49151ea3bfb760b58804eb0d.

vmarta committed:
Merge branch 'feature/extensibility-v2' into vm/clean-up-some-deps

vmarta committed:
Jest now able to find the modules that start with 'overridable!'

vmarta committed:
Update jest config for the storefront too
